tests_integration/test_data_provider/test_databases/test_snowflake.py

```diff
@@ -156,3 +167,109 @@ def test_write_method(dataset_table_fixture):
     rows = dp.fetch_all_rows(table=dp.dataset)
     rows.sort(key=lambda x: x[0])
     assert rows == [(1, "First"), (2, "Second"), (3, "Third with unicode पांचाल")]
+
+
+@pytest.mark.parametrize(
+    "src_dataset_fixture",
+    [
+        {
+            "name": "SnowflakeDataProvider",
+        }
+    ],
+    indirect=True,
+    ids=lambda dp: dp["name"],
+)
+def test_s3_to_snowflake_native_path(sample_dag, src_dataset_fixture):
+    """
+    Test the native path of S3 to Snowflake
+    """
+    table_dataprovider, table_dataset = src_dataset_fixture
+    file_dataset = File(path="s3://astro-sdk/sample.csv", conn_id="aws_default", filetype=FileType.CSV)
+    file_dataprovider = create_dataprovider(file_dataset)
+    with sample_dag:
+        # [START transfer_from_s3_to_snowflake_natively]
+        # file_dataset = File(path="s3://astro-sdk/sample.csv", conn_id="aws_default", filetype=FileType.CSV)
+        # table_dataset = Table(conn_id="snowflake_conn")
+        UniversalTransferOperator(
+            task_id="s3_to_bigquery",
+            source_dataset=file_dataset,
+            destination_dataset=table_dataset,
+            transfer_params={
+                "file_options": {"SKIP_HEADER": 1},
+                "storage_integration": SNOWFLAKE_STORAGE_INTEGRATION_AMAZON,
+            },
+            transfer_mode=TransferMode.NATIVE,
+        )
+        # [END transfer_from_s3_to_snowflake_natively]
+    run_dag(sample_dag)
+    file_dataframe = export_to_dataframe(file_dataprovider)
+    table_dataframe = export_to_dataframe(table_dataprovider)
+    assert file_dataframe.equals(table_dataframe)
+
+
+@pytest.mark.parametrize(
+    "src_dataset_fixture",
+    [
+        {
+            "name": "SnowflakeDataProvider",
+        }
+    ],
+    indirect=True,
+    ids=lambda dp: dp["name"],
+)
+def test_gcs_to_snowflake_native_path(sample_dag, src_dataset_fixture):
+    """
+    Test the native path of GCS to Snowflake
+    """
+    table_dataprovider, table_dataset = src_dataset_fixture
+    file_dataset = File(
+        path="gs://astro-sdk/workspace/sample.csv", conn_id="google_cloud_default", filetype=FileType.CSV
+    )
+    file_dataprovider = create_dataprovider(dataset=file_dataset)
+    with sample_dag:
+        # [START transfer_from_gcs_to_snowflake_natively]
+        # file_dataset = File(
+        #     path="gs://astro-sdk/workspace/sample.csv", conn_id="google_cloud_default", filetype=FileType.CSV
+        # )
+        # table_dataset = Table(conn_id="snowflake_conn")
```

Comment:
```suggestion
```

Reply:
@phanikumv We need this in as this code is used in the doc and without this code comment the example won't be meaningful.